tests/test_transpilers_routing.py

```diff
@@ -0,0 +1,239 @@
+import networkx as nx
+import numpy as np
+import pytest
+from qibo import gates
+from qibo.models import Circuit
+
+from qibolab.transpilers.placer import Custom, Subgraph, Trivial, check_placement
+from qibolab.transpilers.routing import (
+    ShortestPaths,
+    remap_circuit,
+    respect_connectivity,
+)
+
+
+def star_connectivity():
+    Q = ["q" + str(i) for i in range(5)]
+    chip = nx.Graph()
+    chip.add_nodes_from(Q)
+    graph_list = [
+        (Q[0], Q[2]),
+        (Q[1], Q[2]),
+        (Q[3], Q[2]),
+        (Q[4], Q[2]),
+    ]
+    chip.add_edges_from(graph_list)
+    return chip
+
+
+def generate_random_circuit(nqubits, ngates, seed=42):
+    """Generate a random circuit with RX and CZ gates."""
+    np.random.seed(seed)
+    one_qubit_gates = [gates.RX, gates.RY, gates.RZ]
+    two_qubit_gates = [gates.CZ, gates.CNOT, gates.SWAP]
+    n1, n2 = len(one_qubit_gates), len(two_qubit_gates)
+    n = n1 + n2 if nqubits > 1 else n1
+    circuit = Circuit(nqubits)
+    for _ in range(ngates):
+        igate = int(np.random.randint(0, n))
+        if igate >= n1:
+            q = tuple(np.random.randint(0, nqubits, 2))
+            while q[0] == q[1]:
+                q = tuple(np.random.randint(0, nqubits, 2))
+            gate = two_qubit_gates[igate - n1]
+        else:
+            q = (np.random.randint(0, nqubits),)
+            gate = one_qubit_gates[igate]
+        if issubclass(gate, gates.ParametrizedGate):
+            theta = 2 * np.pi * np.random.random()
+            circuit.add(gate(*q, theta=theta, trainable=False))
+        else:
+            circuit.add(gate(*q))
+    return circuit
+
+
+def test_is_satisfied():
+    transpiler = ShortestPaths(connectivity=star_connectivity(), verbose=False)
+    circuit = Circuit(5)
+    circuit.add(gates.CZ(0, 2))
+    circuit.add(gates.CZ(1, 2))
+    circuit.add(gates.Z(1))
+    circuit.add(gates.CZ(2, 1))
+    circuit.add(gates.M(0))
+    assert transpiler.is_satisfied(circuit)
+
+
+@pytest.mark.parametrize("verbose", [True, False])
+def test_respect_connectivity(verbose):
+    circuit = Circuit(5)
+    circuit.add(gates.CZ(0, 2))
+    circuit.add(gates.CZ(1, 2))
+    circuit.add(gates.Z(1))
+    circuit.add(gates.CZ(2, 1))
+    circuit.add(gates.M(0))
```

Comment:
Same circuit of above: make a simple function for this (if it can have a short descriptive name it would be the best!)